### PR TITLE
#4246: Limit width of text body for all themes

### PR DESCRIPTION
**Pull request by timhoffm**
### Feature: Limit width of text body for all themes

Implements #4246 

The purpose of this PR is to limit the text with of the documentation pages to a reasonable size:
- Do not make too long lines, even if the horizontal screen space is very large.
- Do not wrap to arbitrarily small widths (instead add a scrollbar if necessary).

This PR adds `min-width` and `max-width` CSS properties to the body. 

They are configurable via the theme options `minbodywidth` and `maxbodywidth`.

Note: I've named the options without any separating chars as this seems to be the general style for theme options (e.g. `sidebarlinkcolor`). This makes it a bit hard to read and I wouldn't mind renaming to `min_body_width` (also for a more pythonic feeling of `conf.py` `html_theme_options`). Please let me know what the naming conventions for theme options are.

### CODE REVIEW

File: doc/theming.rst [hunk 1]
```diff
@@ -117,6 +117,12 @@ These themes are:
   - **sidebarwidth** (an integer): Width of the sidebar in pixels.  (Do not
     include ``px`` in the value.)  Defaults to 230 pixels.
 
+  - **minbodywidth** (an integer): Minimal width of the document body in
+    pixels.  Defaults to 450 pixels.
+
+  - **minbodywidth** (an integer): Maximal width of the document body in
+    pixels.  Defaults to 800 pixels.
```
Comment (tk0miya) [inline doc/theming.rst]: Does this value also go well with themes not having sidebars (ex. haiku, scrolls and so on)?

**Comment by tk0miya** [comment]
Note: test has been failed because of timed-out. So I restarted it manually and passed now.

**Comment by timhoffm** [comment]
I've refined the PR:

- Options are now `body_min_width` / `body_max_width`. I realized that there are already some options using underscores. So having underscores does not break conventions, and it's more readable. Also I've changed the pattern to `[element]_[css_property]`.
- The new options as well as the existing `sidebarwidth` do now support integer values (interpreted as pixels) as well as any valid CSS dimension string, e.g. '70em'.
- Works now with all themes (except epub, where IMO the option does not make sense - but I'm no expert with epub).

File: doc/theming.rst (hunk 1 above)
Comment (timhoffm) [reply doc/theming.rst]: The refined version does now.

**Comment by tk0miya** [comment]
Thanks!

**Pull request merged**